components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -178,7 +208,28 @@ class DownloadsFeature(
         ).show()
     }
 
-    private fun showDialog(tab: SessionState, download: DownloadState) {
+    @VisibleForTesting
+    internal fun showAppDownloaderDialog(tab: SessionState, download: DownloadState, apps: List<DownloaderApp>) {
+        appChooserDialog.setApps(apps)
+        appChooserDialog.onAppSelected = { app ->
+            if (app.packageName == applicationContext.packageName) {
+                startDownload(download)
+            } else {
+                applicationContext.startActivity(app.toIntent())
```

Comment:
Do we want to wrap that here in case launching the activity failed?

Reply:
👍